src/main/java/frc/robot/subsystems/swerve/swervestatehelpers/SwerveStateHelper.java

```diff
@@ -0,0 +1,73 @@
+package frc.robot.subsystems.swerve.swervestatehelpers;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import frc.robot.constants.MathConstants;
+import frc.robot.subsystems.swerve.Swerve;
+import frc.robot.subsystems.swerve.SwerveConstants;
+import frc.robot.subsystems.swerve.SwerveState;
+import frc.robot.subsystems.swerve.module.ModuleUtils;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+
+public class SwerveStateHelper {
+
+	private final Swerve swerve;
+	private final SwerveConstants swerveConstants;
+	private final Supplier<Optional<Pose2d>> robotPoseSupplier;
+	private final Supplier<Optional<Translation2d>> noteTranslationSupplier;
+
+	public SwerveStateHelper(
+		Supplier<Optional<Pose2d>> robotPoseSupplier,
+		Supplier<Optional<Translation2d>> noteTranslationSupplier,
+		Swerve swerve
+	) {
+		this.swerve = swerve;
+		this.swerveConstants = swerve.getConstants();
+		this.robotPoseSupplier = robotPoseSupplier;
+		this.noteTranslationSupplier = noteTranslationSupplier;
+	}
+
+	public ChassisSpeeds applyAimAssistOnChassisSpeeds(ChassisSpeeds speeds, SwerveState swerveState) {
+		return switch (swerveState.getAimAssist()) {
+			case NONE -> speeds;
+		};
+	}
+
+
+	public Translation2d getRotationAxis(RotateAxis rotationAxisState) {
+		return switch (rotationAxisState) {
+			case MIDDLE_OF_ROBOT -> new Translation2d();
+			case FRONT_LEFT_MODULE -> swerve.getModules().getModule(ModuleUtils.ModulePosition.FRONT_LEFT).getPositionFromCenterMeters();
+			case FRONT_RIGHT_MODULE -> swerve.getModules().getModule(ModuleUtils.ModulePosition.FRONT_RIGHT).getPositionFromCenterMeters();
+			case BACK_LEFT_MODULE -> swerve.getModules().getModule(ModuleUtils.ModulePosition.BACK_LEFT).getPositionFromCenterMeters();
+			case BACK_RIGHT_MODULE -> swerve.getModules().getModule(ModuleUtils.ModulePosition.BACK_RIGHT).getPositionFromCenterMeters();
+		};
+	}
+
+	public RotateAxis getFarRotateAxis(boolean isLeft) {
+		Rotation2d currentAllianceHeading = swerve.getAllianceRelativeHeading();
+		if (Math.abs(currentAllianceHeading.getDegrees()) <= MathConstants.EIGHTH_CIRCLE.getDegrees()) { // -45 <= x <= 45
+			return isLeft ? RotateAxis.FRONT_LEFT_MODULE : RotateAxis.FRONT_RIGHT_MODULE;
+		}
+		if (Math.abs(currentAllianceHeading.getDegrees()) >= MathConstants.EIGHTH_CIRCLE.getDegrees() * 3) { // -135 - x - 135
+			return isLeft ? RotateAxis.BACK_RIGHT_MODULE : RotateAxis.BACK_LEFT_MODULE;
+		}
+		if (currentAllianceHeading.getDegrees() > 0) { // 45 <= x <= 135
+			return isLeft ? RotateAxis.FRONT_RIGHT_MODULE : RotateAxis.BACK_RIGHT_MODULE;
+		}
+		return isLeft ? RotateAxis.BACK_LEFT_MODULE : RotateAxis.FRONT_LEFT_MODULE; // -45 >= x >= -135
+	}
```

Comment:
can you explain to me what does it do and why do we need it? just asking

Reply:
What?: gives the left/right far modules from the driver
Why?: for rotations around the module instead of the debtor of the robot (if we learn how to use it can be good vs defence in games)